src/commands/query.mjs

```diff
@@ -0,0 +1,135 @@
+    } catch (e) {
+      throw new Error(`Unable to write to output directory: ${outputDir}`);
+    }
+  }
+}
+
+const resolveInput = (argv) => {
+  const { readFileSync } = container.resolve("fs");
+  const logger = container.resolve("logger");
+
+  if (argv.input) {
+    logger.debug(`reading query from ${argv.input}`, "argv");
+    return readFileSync(argv.input, "utf8");
+  }
+
+  if (argv.fql === "-") {
+    logger.debug("reading query from stdin", "argv");
+    return readFileSync(process.stdin.fd, "utf8");
+  }
+
+  logger.debug("no --input specified, using [fql]", "argv");
+  return argv.fql;
+}
+
+async function queryCommand(argv) {
+  // validate the arguments and throw if they are invalid
+  validateDatabaseOrSecret(argv);
+  validate(argv);
+
+  // resolve the input
+  const expression = resolveInput(argv);
+
+  // get the query handler and run the query
+  try {
+    const secret = await getSecret();
+    const { url, timeout, typecheck, extra, json, apiVersion } = argv
+    const results = await container.resolve("runQueryFromString")(expression, {
+      apiVersion,
+      secret,
+      url,
+      timeout,
+      typecheck,
+    });
+    const output = formatQueryResponse(results, {
+      apiVersion,
+      extra,
+      json,
+    });
+
+    if (argv.output) {
+      container.resolve("fs").writeFileSync(argv.output, output);
+    } else {
+      container.resolve("logger").stdout(output);
+    }
+
+    return results;
+  } catch (err) {
+    err.message = formatError(err, { apiVersion: argv.apiVersion, extra: argv.extra }); 
+    throw err;
+  }
+}
+
+function buildQueryCommand(yargs) {
+  return yargs
+    .positional("fql", {
+      type: "string",
+      description: "the query to run; use - to read from stdin",
+    })
+    .nargs('fql', 1)
+    .options({
+      input: {
+        alias: "i",
+        type: "string",
+        description:
+          "file path to read the query (or queries) from",
+      },
+      output: {
+        alias: "o",
+        type: "string",
+        description: "file path to write output to; defaults to stdout",
+      },
+      extra: {
+        type: "boolean",
+        description: "include additional information in the output, including stats",
+        default: false,
+      },
+      ...commonConfigurableQueryOptions,
+    })
+    .example([
+      ['$0 query "Collection.all()" --database us-std/example --role admin', "run the query and write to stdout "],
+      ["$0 query -i /path/to/queries.fql --database us-std/example --role admin", "run the query from a file"],
+      ['echo "1 + 1" | $0 query - --database us-std/example --role admin', "run the query from stdin"],
+      ['$0 query -i /path/to/queries.fql -o /tmp/result.json --database us-std/example --role admin', "run the query and write to a file"],
+      ['$0 query -i /path/to/queries.fql -o /tmp/result.json --extra --database us-std/example --role admin', "run the query and write full API response to a file"],
```

Comment:
--role will default to admin when minting a key for the specified database path. Maybe examples should show you don't have to provide role arg.

Also reminds me to go test out custom user roles...